### DISCUSSION TITLE
Changing the value of a Controlled MUI field

### DISCUSSION THREAD

**Discussion by MicheleGrisafi** [react-hook-form/react-hook-form#8980] [Q&A]
Hi, I'm stuck with a problem that I cannot solve on my own. I am using both react-hook-form and MUI. Following the RHF guidelines, I'm wrapping my MUI TextInput in the Controller element. However, I also need to change the value of the fields manually, and the guidelines say that it cannot be done with setValue(). What should I do?
Here is the component:

   ```
 <Controller 
          control={control}
          name={inputName}
          defaultValue={defaultValue}
          render={({field:{onChange, value}})=>(
            <TextField 
              label={labelText}
              onChange={onChange}
              value={value}
              />
          )}
        />
```

**Comment by Moshyfawn** [comment]
Can you provide a use-case for changing the values manually and reference the docs part, that says that you shouldn't use `setValue` for this?

**Reply by MicheleGrisafi** [reply]
Well, the use case is for instance the following: with the click of a button I want to set a certain value to the field, without needing to type it.
As for the docs, the page is [this](https://react-hook-form.com/api/usecontroller/controller), and you can simply search the page for "setValue". I quote "This prop update [formState](https://react-hook-form.com/api/useform/formstate) and you should avoid manually invoke [setValue](https://react-hook-form.com/api/useform/setvalue) or other API related to field update.".

**Comment by leapful** [comment]
@MicheleGrisafi : I think you are misunderstanding the documentation. The docs warns about avoiding calling "setValue" or any API that update field value along with "field.onChange".

For example:

```
const SomeCustomComponent = ({ setValue }) => {
   const { field } = useController();

   return <input name="custom" onChange={() => {
         field.onChange(value);
         setValues('custom', value);
    }} />
}
```

Beside that, you can freely use API "setValue" to manually update field value anywhere outside its controller.
